docs/src/pages/guides/api/api.md

```diff
@@ -136,3 +136,10 @@ The Material-UI components use a combination of the two approaches according to
 - An *enum* is used when **> 2** degrees of freedom are required, or if there is the possibility that additional degrees of freedom may be required in the future.
 
 Going back to the previous button example; since it requires 3 degrees of freedom, we use an *enum*.
+
+## Glossary
+
+- **BFS**: Breadth-first search.
+- **host element**: a DOM node in the context of `react-dom`.
+- **root component**: the outermost (first when doing BFS) component that renders a host element.
+- **root element**: the outermost (first when doing BFS) element that renders a host element.
```

Comment:
I would just go with root component. We don't need both terms. Not sure if there's a distinction between an element rendering something and a component. I would think the component does the rendering not the element. Elements are just a description of the UI.

Reply:
We can take the example of the Card. The root component of the Card is the Paper, the root element of the Card is a div. Correct?

Reply:
I wouldn't introduce another definition of elements since element is already used in React. Where would we use this currently?

Reply:
Your argument is invalid. I wouldn't introduce another definition of components since component is already used in React.

In the documentation, we are saying:
> Any other properties supplied will be spread to the root element (native element).

You don't spread props to a component, you spread them to a component instance: an element. My previous comment is wrong. I should have said:

> We can take the example of the Card. The root component of the Card is the Paper, the root element of the Card is a Paper element. The host instance is a div.

Reply:
Maybe we should only give a definition to `root`?

Reply:
> Your argument is invalid. I wouldn't introduce another definition of components since component is already used in React.

I think we have a misunderstanding. I don't want to remove the root component point but root element. I don't know why we need both.

> You don't spread props to a component, you spread them to a component instance: an element. 

elements are not instances though. I thought I gave a sufficient interpretation of the difference in https://github.com/mui-org/material-ui/pull/15337#discussion_r275119715. This is included in the article you linked in that convo.

Reply:
Maybe we should ask differently: To what *thing* do we give props? To the component or the element? Whichever is the answer also means it's the target of the spread.

Reply:
We give them to the element.

Reply:
> Maybe we should only give a definition of root?

That's more helpful. Everything else will just add more confusion depending on whether people use to say they spread props to the element or the component.

Reply:
Actually, I think that by providing two different definitions for root element and root component we make it clear that they are two important different concepts.